components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -157,14 +169,114 @@ public void setUp() throws Exception {
 
     }
 
-    @Test
-    public void testBuildIDToken() throws Exception {
+    @DataProvider(name = "TestBuildRequestIDToken")
+    public Object[][] buildRequestIDToken() throws Exception {
+        // second set of values
+        OAuth2AccessTokenReqDTO tokenReqDTO2 = new OAuth2AccessTokenReqDTO();
+        tokenReqDTO2.setTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+        tokenReqDTO2.setClientId("dabfba9390aa423f8b04332794d83614");
+        tokenReqDTO2.setCallbackURI(TestConstants.CALLBACK);
+        OAuthTokenReqMessageContext messageContext2 = new OAuthTokenReqMessageContext(tokenReqDTO2);
+        messageContext2.setAuthorizedUser(user);
+        messageContext2.setScope(TestConstants.OPENID_SCOPE_STRING.split(" "));
+        messageContext2.addProperty(AUTHORIZATION_CODE, AUTHORIZATION_CODE_VALUE);
+
+        OAuth2AccessTokenRespDTO tokenRespDTO2 = new OAuth2AccessTokenRespDTO();
+        tokenRespDTO2.setAccessToken("2sa9a678f890877856y66e75f605d456");
+        AuthenticatedUser user2 = new AuthenticatedUser();
+        user2.setAuthenticatedSubjectIdentifier("user2");
+        user2.setUserName("user2");
+        user2.setUserStoreDomain(TestConstants.USER_STORE_DOMAIN);
+        user2.setTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+        user2.setFederatedUser(true);
+        messageContext2.setAuthorizedUser(user2);
+
+        return new Object[][]{
+                {messageContext, tokenRespDTO, TestConstants.CLIENT_ID},
+                {messageContext2, tokenRespDTO2, "dabfba9390aa423f8b04332794d83614"}
+        };
+    }
+
+    @Test(dataProvider = "TestBuildRequestIDToken")
+    public void testBuildIDToken(OAuthTokenReqMessageContext messageContext,
+                                 OAuth2AccessTokenRespDTO tokenRespDTO, String clientId) throws Exception {
+
+        RealmService realmService = IdentityTenantUtil.getRealmService();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
+        IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
+        String idToken = defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getAudience().get(0),
+                clientId);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getIssuer(),
+                "https://localhost:9443/oauth2/token");
+        }
+
+    @DataProvider(name = "TestBuildAuthzIDToken")
+    public Object[][] buildAuthzIDToken() throws Exception {
+        OAuth2AuthorizeReqDTO authzTokenReqDTO = new OAuth2AuthorizeReqDTO();
```

Comment:
Formatting: The 1st line of the method should be an empty line. Check and fix in other places also